android/src/main/java/com/appboy/reactbridge/AppboyReactBridge.java

```diff
@@ -599,6 +616,174 @@ public void getInstallTrackingId(Callback callback) {
     reportResultWithCallback(callback, null, Appboy.getInstance(getReactApplicationContext()).getInstallTrackingId());
   }
 
+  @ReactMethod
+  public void getContentCards(final Promise promise) {
+    final IEventSubscriber<ContentCardsUpdatedEvent> subscriber = new IEventSubscriber<ContentCardsUpdatedEvent>() {
+      @Override
+      public void trigger(ContentCardsUpdatedEvent event) {
+        promise.resolve(mapContentCards(event.getAllCards()));
+        updateContentCardsIfNeeded(event);
+        Appboy.getInstance(getReactApplicationContext()).removeSingleSubscription(this, ContentCardsUpdatedEvent.class);
+      }
+    };
+    Appboy.getInstance(getReactApplicationContext()).subscribeToContentCardsUpdates(subscriber);
+    Appboy.getInstance(getReactApplicationContext()).requestContentCardsRefresh(true);
+  }
+
+  private WritableArray mapContentCards(List<Card> cardsList) {
+    WritableArray cards = Arguments.createArray();
+    for (Card card : cardsList.toArray(new Card[0])) {
+      cards.pushMap(mapContentCard(card));
+    }
+    return cards;
+  }
+
+  private WritableMap mapContentCard(Card card) {
+    WritableMap mappedCard = Arguments.createMap();
+    mappedCard.putString("id", card.getId());
```

Comment:
Ideally, these keys should live under a/some symbolic constant(s), not all of them have a single occurrence.

Reply:
updated